### ISSUE TITLE
correlogram

### ISSUE THREAD

**Issue by knaaptime** [pysal/esda#252]
I needed a correlogram for a project i'm working on but noticed we didnt have one hand in the library, so cooked this one up. It's roughly equivalent to the [R version](https://r-spatial.github.io/spdep/reference/sp.correlogram.html#references-1) (though I notice [geoda only implements](0http://geodacenter.github.io/workbook/5a_global_auto/lab5a.html#spatial-correlogram) the non-parametric version)

if i add some tests, do we want it here?

https://gist.github.com/knaaptime/510fd5f03fed425f3d3d5163b599320d

**Comment by ljwolf** [comment]
this would be similarly benefitted by the incremental KDTree in #125... I think we should totally include it in `esda`!

**Comment by knaaptime** [comment]
yeah i was letting it linger, but seeing that activity made it clear its worth raising :)

**Comment by edurojasr** [comment]
+1 to add this in PySAL

I'm really new on using PySAL and so far I'm love it! One of the first thing I needed to do was the correlogram thanks for all the work PySAL is great!, maybe this is not the best place to ask, but do you know of good resources to learn Pysal? books or preferably courses or certifications, I thank you for the guide you can give me

**Comment by jGaboardi** [comment]
Resolved in https://github.com/pysal/esda/pull/259
